### PR TITLE
Added 'things to watch out for' section in Kubernetes instructions

### PR DESCRIPTION
**Pull request by bboreham**
Triggered by #3127 

I typed it in a bit of a hurry; am sure the English can be improved

### CODE REVIEW

File: site/kubernetes/kube-addon.md [hunk 1]
```diff
@@ -270,6 +271,14 @@ TCP connection from 10.32.0.7:56648 to 10.32.0.11:80 blocked by Weave NPC.
 UDP connection from 10.32.0.7:56648 to 10.32.0.11:80 blocked by Weave NPC.
 ```
 
+### <a name="key-points"></a><a name="key-points"></a> Things to watch out for
+
+- Don't turn on `--masquerade-all` on kube-proxy: this will change the
+  source address of every pod-to-pod conversation which will make it
+  impossible to write network policies that restrict which pods can talk.
```
Comment (rade) [inline site/kubernetes/kube-addon.md]: "write" -> "correctly enforce"

File: site/kubernetes/kube-addon.md [hunk 2]
```diff
@@ -270,6 +271,14 @@ TCP connection from 10.32.0.7:56648 to 10.32.0.11:80 blocked by Weave NPC.
 UDP connection from 10.32.0.7:56648 to 10.32.0.11:80 blocked by Weave NPC.
 ```
 
+### <a name="key-points"></a><a name="key-points"></a> Things to watch out for
```
Comment (mikebryant) [inline site/kubernetes/kube-addon.md]: nit: Why is the `<a name="key-points"></a>` repeated?

Comment (bboreham) [reply site/kubernetes/kube-addon.md]: it was pasted from `troubleshooting` without going through the conscious mind...

File: site/kubernetes/kube-addon.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ The following topics are discussed:
 * [Network Policy Controller](#npc)
 * [Troubleshooting](#troubleshooting)
  * [Troubleshooting Blocked Connections](#blocked-connections)
+ * [Things to watch out for](#key-points)
 * [Changing Configuration Options](#configuration-options)
```
Comment (abuehrle) [inline site/kubernetes/kube-addon.md]: change to:  
   * Connection Options to be Aware of

(also list item should be indented as a child of Troubleshooting Blocked Connections)

File: site/kubernetes/kube-addon.md [hunk 4]
```diff
@@ -270,6 +271,15 @@ TCP connection from 10.32.0.7:56648 to 10.32.0.11:80 blocked by Weave NPC.
...
+### <a name="key-points"></a> Things to watch out for
+
+- Don't turn on `--masquerade-all` on kube-proxy: this will change the
+  source address of every pod-to-pod conversation which will make it
+  impossible to correctly enforce network policies that restrict which
+  pods can talk.
+- If you do set the `--cluster-cidr` option on kube-proxy, make sure
+  it matches the `IPALLOC_RANGE` given to Weave Net (see below)
+
 ## <a name="configuration-options"></a> Changing Configuration Options
```
Comment (abuehrle) [inline site/kubernetes/kube-addon.md]: Don't use `--masquerade-all` on kube-proxy: this setting changes the
source address of every pod-to-pod conversation which makes it
impossible to correctly enforce the network policies that restrict which
pods can communicate.

If you set the `--cluster-cidr` option on kube-proxy, ensure
it matches the `IPALLOC_RANGE` given to Weave Net (see below)

File: site/kubernetes/kube-addon.md (hunk 3 above)
Comment (bboreham) [reply site/kubernetes/kube-addon.md]: The list is not about connection options; it's all the things we want to tell users about Weave Net on Kubernetes.

File: site/kubernetes/kube-addon.md (hunk 4 above)
Comment (bboreham) [reply site/kubernetes/kube-addon.md]: I don't think "the network policies" gives the right meaning - there are no policies by default; only ones you write, and you have to write ones which seek to restrict pods before you hit the problem (you can still write ones that restrict ports, for instance)

**Pull request merged**